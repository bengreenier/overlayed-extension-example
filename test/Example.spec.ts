import React from "react"
import renderer from "react-test-renderer"
import Example from "../src/index"

// test the Example component
describe("Example", () => {
  // tests that you must run to ensure overlayed can handle your plugin at runtime
  //
  // TODO(bengreenier): make these easier to consume
  describe("overlayed mandated tests", () => {
    // this test ensures that overlayed can load the plugin
    test("loads safely", () => {
      return expect(import("../src/index")).resolves.toBeDefined()
    })

    // this test ensures that overlayed can render the plugin
    test("renders safely", () => {
      expect(renderer.create(React.createElement(Example))).toBeDefined()
    })
  })

  // this test does snapshot testing to ensure the component renders properly
  // see https://jestjs.io/docs/en/tutorial-react for more info
  test("renders greeting", () => {
    // render the Example component, with default props
    let comp = renderer.create(React.createElement(Example))

    // we use the 'as' cast to avoid nullchecks below
    let tree = comp.toJSON() as renderer.ReactTestRendererJSON

    // snapshot test default props
    expect(tree).toMatchSnapshot()

    // render the Example component, with given props
    comp = renderer.create(React.createElement(Example, { username: "test" }))
    tree = comp.toJSON() as renderer.ReactTestRendererJSON

    // snapshot test given props
    expect(tree).toMatchSnapshot()
  })
})
