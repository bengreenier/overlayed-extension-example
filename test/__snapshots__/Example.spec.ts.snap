// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example renders greeting 1`] = `
<h1>
  Hello human
</h1>
`;

exports[`Example renders greeting 2`] = `
<h1>
  Hello test
</h1>
`;
